src/heretic/model.py

```diff
@@ -63,11 +66,38 @@ def __init__(self, settings: Settings):
             print(f"* Trying dtype [bold]{dtype}[/]... ", end="")
 
             try:
+                quantization_config = None
+                if settings.quantization == QuantizationMethod.BNB_4BIT:
+                    # BitsAndBytesConfig expects a torch.dtype, not a string.
+                    if dtype == "auto":
+                        compute_dtype = torch.bfloat16
+                    else:
+                        compute_dtype = getattr(torch, dtype)
+
+                    quantization_config = BitsAndBytesConfig(
+                        load_in_4bit=True,
+                        bnb_4bit_compute_dtype=compute_dtype,
+                        bnb_4bit_quant_type="nf4",
+                        bnb_4bit_use_double_quant=True,
+                    )
+                elif settings.quantization == QuantizationMethod.BNB_8BIT:
+                    quantization_config = BitsAndBytesConfig(
+                        load_in_8bit=True,
+                    )
+
+                # Build kwargs, only include quantization_config if it's not None
+                # (some models like gpt-oss have issues with explicit None)
+                load_kwargs = {
+                    "dtype": dtype,
+                    "device_map": settings.device_map,
+                    "trust_remote_code": self.trusted_models.get(settings.model),
```

Comment:
These arguments can be kept in the function call, only `quantization_config` is dynamic. Keeping them there gives us proper typing,